src/System.CommandLine/Help/HelpBuilder.Default.cs

```diff
@@ -50,63 +50,46 @@ public static string GetArgumentDefaultValue(Argument argument)
         /// </summary>
         public static string GetArgumentUsageLabel(Argument argument)
         {
-            if (argument.ValueType == typeof(bool) ||
-                argument.ValueType == typeof(bool?))
+            // Argument.HelpName is always first choice
+            if (!string.IsNullOrWhiteSpace(argument.HelpName))
             {
-                if (argument.FirstParent?.Symbol is Command)
-                {
-                    return $"<{argument.Name}>";
-                }
-                else
-                {
-                    return "";
-                }
+                return $"<{argument.HelpName}>";
             }
+            else if (!argument.IsBoolean() && argument.CompletionSources.Count > 0)
```

Comment:
this change fixes #2038

First of all we check `HelpName`, if it's not provided and there are some `CompletionSources` defined for given `Argument`, then we are getting the completions.